File: DotNet/CesiumLanguageWriter/Advanced/CesiumDoubleValuePropertyAdaptor.cs
using JetBrains.Annotations;

namespace CesiumLanguageWriter.Advanced
{
    /// <summary>
    /// Adapts a class that implements <see cref="ICesiumDoubleValuePropertyWriter"/> to implement
    /// <see cref="ICesiumInterpolatableValuePropertyWriter{TValue}"/> for <see cref="double"/> values.
    /// </summary>
    /// <typeparam name="TFrom">The class that implements <see cref="ICesiumDoubleValuePropertyWriter"/> to adapt.</typeparam>
    public class CesiumDoubleValuePropertyAdaptor<TFrom> : CesiumInterpolatableWriterAdaptor<TFrom, double>
        where TFrom : class, ICesiumDoubleValuePropertyWriter
    {
        /// <summary>
        /// Initializes a new instance.
        /// </summary>
        /// <param name="parent">The instance to wrap.</param>
        /// <param name="writeValueCallback">The callback to write values of type <see cref="double"/>.</param>
        /// <param name="writeSamplesCallback">The callback to write samples of type <see cref="double"/>.</param>
        /// <param name="writeDeleteValueCallback">The callback to write an indication that the client should delete existing data.</param>
        public CesiumDoubleValuePropertyAdaptor([NotNull] TFrom parent,
                                                [NotNull] CesiumWriterAdaptorWriteCallback<TFrom, double> writeValueCallback,
                                                [NotNull] CesiumWriterAdaptorWriteSamplesCallback<TFrom, double> writeSamplesCallback,
                                                [NotNull] CesiumWriterAdaptorWriteDeleteCallback<TFrom> writeDeleteValueCallback)
            : base(parent, writeValueCallback, writeSamplesCallback, writeDeleteValueCallback)
        {
        }
    }
}

File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/advanced/CesiumDoubleValuePropertyAdaptor.java
package cesiumlanguagewriter.advanced;


import agi.foundation.compatibility.*;
import cesiumlanguagewriter.*;
import javax.annotation.Nonnull;

/**
 * Adapts a class that implements {@link ICesiumDoubleValuePropertyWriter} to implement
 {@link ICesiumInterpolatableValuePropertyWriter} for {@code double} values.
 * @param <TFrom> The class that implements {@link ICesiumDoubleValuePropertyWriter} to adapt.
 */
@SuppressWarnings( {
        "unused",
        "deprecation",
        "serial"
})
public class CesiumDoubleValuePropertyAdaptor<TFrom extends ICesiumDoubleValuePropertyWriter> extends CesiumInterpolatableWriterAdaptor<TFrom, Double> {
    /**
    * Initializes a new instance.
    * @param parent The instance to wrap.
    * @param writeValueCallback The callback to write values of type {@code double}.
    * @param writeSamplesCallback The callback to write samples of type {@code double}.
    * @param writeDeleteValueCallback The callback to write an indication that the client should delete existing data.
    */
    public CesiumDoubleValuePropertyAdaptor(@Nonnull TFrom parent, @Nonnull CesiumWriterAdaptorWriteCallback<TFrom, Double> writeValueCallback,
            @Nonnull CesiumWriterAdaptorWriteSamplesCallback<TFrom, Double> writeSamplesCallback, @Nonnull CesiumWriterAdaptorWriteDeleteCallback<TFrom> writeDeleteValueCallback) {
        super(parent, writeValueCallback, writeSamplesCallback, writeDeleteValueCallback);
    }
}